core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2134,10 +2141,13 @@ class ReplicaManager(val config: KafkaConfig,
             stateChangeLogger.trace(s"Unable to start fetching $tp with topic " +
               s"ID ${info.topicId} because the replica manager is shutting down.")
           } else {
-            val leader = info.partition.leader
-            if (newImage.cluster.broker(leader) == null) {
+            if (isInControlledShutdown && !info.partition.isr.contains(config.brokerId)) {
```

Comment:
I wonder if we should also count the case when the replica is in the ISR, but no leader is defined. I think this is what we would see if the shutting down replica is the last member of the ISR.

Reply:
That makes sense. I forgot about this case.